hnn_core/tests/test_network.py

```diff
@@ -28,3 +31,51 @@ def test_network():
     for ev_input in params['t_ev*']:
         type_key = ev_input[2: -2] + ev_input[-1]
         assert len(net.gid_dict[type_key]) == net.n_cells
+
+    # Assert that an empty Spikes object is created as an attribute
+    assert net.spikes == Spikes()
+
+
+def test_spikes():
+    '''Test spikes object.'''
+
+    # Round-trip test
+    spiketimes = [[2.3456, 7.89], [4.2812, 93.2]]
+    spikegids = [[1, 3], [5, 7]]
+    spiketypes = [['L2_pyramidal', 'L2_basket'], ['L5_pyramidal', 'L5_basket']]
+    spikes = Spikes(times=spiketimes, gids=spikegids, types=spiketypes)
+    spikes.write('/tmp/spk_%d.txt')
+    assert spikes == read_spikes('/tmp/spk_*.txt')
+
+    # TypeError should be raised when one of the args is entered as a non-list
+    # (e.g., when times is a tuple of lists)
+    with pytest.raises(TypeError) as excinfo:
+        spiketimes_tuple = ([2.3456, 7.89], [4.2812, 93.2])
+        spikes = Spikes(times=spiketimes_tuple, gids=spikegids,
+                        types=spiketypes)
+    assert "times should be a list of lists" in str(excinfo.value)
+
+    # TypeError should be raised when one of the args is entered as a list of
+    # non-lists (e.g., when times is a list of ints)
+    with pytest.raises(TypeError) as excinfo:
+        spikes = Spikes(times=[1, 2], gids=spikegids, types=spiketypes)
+    assert "times should be a list of lists" in str(excinfo.value)
+
+    # ValueError should be raised when one of the args is entered as an
+    # incongruent number of trials (e.g., 1 trial when all other args have 2)
```

Comment:
avoid too many comments, it's bad practice (cf [Clean code](https://www.investigatii.md/uploads/resurse/Clean_Code.pdf)). Code should be written so that it can be read with comments adding only additional information.